internal/querynodev2/tasks/concurrent_safe_scheduler.go

```diff
@@ -0,0 +1,187 @@
+package tasks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/milvus-io/milvus/internal/querynodev2/collector"
+	"github.com/milvus-io/milvus/pkg/log"
+	"github.com/milvus-io/milvus/pkg/metrics"
+	"github.com/milvus-io/milvus/pkg/util/conc"
+	"github.com/milvus-io/milvus/pkg/util/merr"
+	"github.com/milvus-io/milvus/pkg/util/metricsinfo"
+	"github.com/milvus-io/milvus/pkg/util/paramtable"
+	"go.uber.org/atomic"
+	"go.uber.org/zap"
+)
+
+func newConcurrentSafeScheduler(policy schedulePolicy) Scheduler {
+	maxReadConcurrency := paramtable.Get().QueryNodeCfg.MaxReadConcurrency.GetAsInt()
+	log.Info("query node use concurrent safe scheduler", zap.Int("max_concurrency", maxReadConcurrency))
+	return &concurrentSafeScheduler{
+		policy:           policy,
+		addChan:          make(chan addTaskReq),
```

Comment:
this channel should be buffered as that provides more chances to merge tasks, an optimization could be like scheduler.go: 104

Reply:
I have add buffer channel for new scheduler in new change,
There is the testing result:
test setup:
- 10 collection
- 768dim*100000 for every collection.
- search param:
  - nq: 15
  - max_concurrency: 50 
  - rate: 50

master implementation:
- average grouping nq: 180
- average latency of search: 1.4s

implementation of this PR:
- average grouping nq: 187
- average latency of search: [1.28, 1.57]

The average grouping nq metric is greatly related to the system load.
I this that the current implement of fifo scheduler is equal to the master version.